### PR TITLE
chore: move module specifier to //core

### PR DESCRIPTION
**Pull request by bartlomieju**
Follow up to: #2493 

Moves `ModuleSpecifier` to `//core/modules.rs`

### CODE REVIEW

File: core/modules.rs [hunk 1]
```diff
@@ -278,6 +282,87 @@ impl<L: Loader> Future for RecursiveLoad<L> {
   }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+#[allow(dead_code)]
+/// Resolved module specifier
+pub struct ModuleSpecifier(Url);
+
+impl ModuleSpecifier {
```
Comment (ry) [inline core/modules.rs]: I think this was nicer in its own file core/modules_specifer.rs

File: core/modules.rs [hunk 2]
```diff
@@ -48,10 +49,13 @@ pub trait Loader: Send + Sync {
     specifier: &str,
     referrer: &str,
     is_root: bool,
-  ) -> Result<String, Self::Error>;
+  ) -> Result<ModuleSpecifier, Self::Error>;
```
Comment (ry) [inline core/modules.rs]: Yeah... I think this is quite reasonable.

File: core/modules.rs [hunk 3]
```diff
@@ -48,10 +49,13 @@ pub trait Loader: Send + Sync {
...
 
-  /// Given an absolute url, load its source code.
-  fn load(&self, url: &str) -> Box<SourceCodeInfoFuture<Self::Error>>;
+  /// Given ModuleSpecifier, load its source code.
+  fn load(
+    &self,
+    module_specifier: &ModuleSpecifier,
+  ) -> Box<SourceCodeInfoFuture<Self::Error>>;
```
Comment (ry) [inline core/modules.rs]: 👍 I think adopting URL at this layer is simplifying.

As long as core/isolate.rs doesn't depend on URL.

File: core/modules.rs (hunk 1 above)
Comment (bartlomieju) [reply core/modules.rs]: You're right :+1: moved

**Pull request merged**